### PR TITLE
[CORDA-1638]: Node crashes in --initial-registration polling mode if doorman returns a transient HTTP error (fix).

### PR DESCRIPTION
**Pull request by sollecitom**
https://r3-cev.atlassian.net/browse/CORDA-1638

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt [hunk 1]
```diff
@@ -160,12 +164,22 @@ open class NetworkRegistrationHelper(private val config: SSLConfiguration,
...
-            if (certificates != null) {
-                return certificates
+            try {
+                val (pollInterval, certificates) = certService.retrieveCertificates(requestId)
+                if (certificates != null) {
+                    return certificates
+                }
+                Thread.sleep(pollInterval.toMillis())
+            } catch (e: ServiceUnavailableException) {
+                idlePeriodDuration = computeNextIdleDoormanConnectionPollInterval.invoke(idlePeriodDuration)
```
Comment (patrickkuo) [inline node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt]: i think `computeNextIdleDoormanConnectionPollInterval` is too long ... maybe just `nextIdleDuration`?  Also can remove invoke and use () directly

File: node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt [hunk 2]
```diff
@@ -254,4 +271,21 @@ class NodeRegistrationHelper(private val config: NodeConfiguration, certService:
...
+
+        init {
+            require(times > 0)
+        }
+
+        private var counter = times
+
+        override fun invoke(@Suppress("UNUSED_PARAMETER") previousPeriod: Duration?): Duration? {
+            synchronized(this) {
+                val nextPeriod = if (counter > 0) period else null
```
Comment (patrickkuo) [inline node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt]: can simplify to ` return if (counter-- > 0) period else null` ?

**Comment by sollecitom** [comment]
@patrickkuo , fixed. I'll wait for the tests to finish before pushing.

File: node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt (hunk 2 above)
Comment (sollecitom) [reply node/src/main/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelper.kt]: Not sure it's "simpler", but changed anyway.

**Pull request merged**